algorithms/src/snark/marlin/ahp/matrices.rs

```diff
@@ -40,21 +38,38 @@ pub(crate) fn to_matrix_helper<F: Field>(matrix: &[Vec<(F, VarIndex)>], num_inpu
     cfg_iter!(matrix)
         .map(|row| {
             let mut row_map = BTreeMap::new();
-            row.iter().for_each(|(fe, column)| {
-                let column = match column {
-                    VarIndex::Public(i) => *i,
-                    VarIndex::Private(i) => num_input_variables + i,
-                };
-                *row_map.entry(column).or_insert_with(F::zero) += *fe;
+            row.iter().for_each(|(val, column)| {
+                if *val != F::zero() {
```

Comment:
```suggestion
                if !val.is_zero() {
```

Reply:
8907510b8

Just FYI, this inspired: https://github.com/AleoHQ/snarkVM/pull/1796
And depending on the response to that PR I'd prefer to `ensure!` this doesn't  happen.